3/permissions.js

```diff
@@ -0,0 +1,29 @@
+//import { rule, and, or, not, shield } from 'graphql-shield'
+const { rule, and, or, not, shield } = require('graphql-shield');
+const jwt = require('jsonwebtoken');
+const secret = require('./secret.js');
+
+const isAuthenticated = rule()(async (parent, args, ctx, info) => {
+  // Is there a Grocer with such email in our database (Prisma)?
+  //console.log(ctx);
+  var valid = false;
+  try{
+	  var payload = jwt.verify(ctx.token, secret);
```

Comment:
I would decode that in `context` and return the `payload`. In this rule you could simply check if `payload != null`.